staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -750,6 +750,12 @@ type APIResource struct {
 	SingularName string `json:"singularName" protobuf:"bytes,6,opt,name=singularName"`
 	// namespaced indicates if a resource is namespaced or not.
 	Namespaced bool `json:"namespaced" protobuf:"varint,2,opt,name=namespaced"`
+	// group is the preferred group of the resource.  Empty implies the group of the containing resource list.
```

Comment:
> I assume that the preferred group must match the preferred version?

I think this group and version should always match and be the preferred one (the one that most people will use.  Our internal code already handles this case today.



> Don't we expect there times when there are multiple versions of the APIResource?

Eventually, yes.  Right now our code doesn't handle the condition and explicitly forbids it.  I think that we should reflect what we have now (and realistically for the next few releases).



> If we want to switch groups then there could even be multiple groups? Would preferred be the most stable?

I think we would handle this as we've done for our preferred versions.  We would add an array of possible GroupVersions and say that they are in the preferred ordering, leaving the choice to the individual api server.